python/ray/serve/_private/replica.py

```diff
@@ -99,6 +100,11 @@ async def __init__(
                 component_name=deployment_name,
                 component_id=replica_tag,
             )
+            self._memory_logger = configure_component_memory_logger(
```

Comment:
For my knowledge, why do we need to hold the `self._memory_logger`? (worth adding the comment in the `configure_component_memory_logger` func)

Reply:
I actually don't think we do need to hold it. This seemed to work even if it went out of scope. Would it be clearer if I removed it?

Reply:
okay, in this case we don't need to return anything. Yeah, from the caller perspective, it is much cleaner :)

Reply:
Sounds good, I removed the `return`.